src/main/java/com/puppycrawl/tools/checkstyle/api/LocalizedMessage.java

```diff
@@ -283,6 +283,10 @@ public LocalizedMessage(
                 sourceClass, customMessage);
     }
 
+    /**
+     * Indicates whether some other object is "equal to" this one.
+     * @noinspection EqualsCalledOnEnumConstant
```

Comment:
we need to explain suppressions in javadoc, unfortunate side effect.
We always keep brief reason of suppression in code or to keep link to web, with detailed explanation. unfortunately here we will damage a big our javadoc, but is better than keep such explanation in code

Reply:
Done.